src/user/user.service.ts

```diff
@@ -13,7 +13,12 @@ export class UserService {
         @InjectModel(User) private readonly model: ReturnModelType<typeof User>,
     ) {}
 
-    find(): Promise<User[]> {
+    find(filter?: Partial<User>): Promise<User[]> {
+        if (filter) {
+            return this.model
+                .find(filter)
+                .exec();
+        }
         return this.model.find().exec();
     }
```

Comment:
The filter should be `any` type since there are lots of filters you can use (further detail in mongoDB official document). Also, there is no need to `if` there since `find(undefined)` behaves like `find()`.

Reply:
copy